services/api/db/models.py

```diff
@@ -0,0 +1,79 @@
+from django.db import models
+from django.contrib.auth.models import BaseUserManager,AbstractBaseUser
+
+#  Custom User Manager
+class UserManager(BaseUserManager):
+  def create_user(self, email, name, password=None, password2=None):
+      """
+      Creates and saves a User with the given email, name and password.
+      """
+      if not email:
+          raise ValueError('User must have an email address')
+
+      user = self.model(
+          email=self.normalize_email(email),
+          name=name,
+      )
+
+      user.set_password(password)
+      user.save(using=self._db)
+      return user
+
+  def create_superuser(self, email, name, password=None):
+      """
+      Creates and saves a superuser with the given email, name and password.
+      """
+      user = self.create_user(
+          email,
+          password=password,
+          name=name,
+      )
+      user.is_admin = True
+      user.save(using=self._db)
+      return user
+
+
+AUTH_PROVIDERS = {'github': 'github', 'google': 'google', 'email': 'email'}
+
+
+#  Custom User Model
+class User(AbstractBaseUser):
+  email = models.EmailField(
+      verbose_name='Email',
+      max_length=255,
+      unique=True,
+  )
+  name = models.CharField(max_length=200)
+  is_verified = models.BooleanField(default=False)
+  is_active = models.BooleanField(default=True)
+  is_admin = models.BooleanField(default=False)
+  created_at = models.DateTimeField(auto_now_add=True)
+  updated_at = models.DateTimeField(auto_now=True)
+  auth_provider = models.CharField(
+      max_length=255, blank=False,
+      null= False, default=AUTH_PROVIDERS.get('email')
+  )
+
+  objects = UserManager()
+
+  USERNAME_FIELD = 'email'
+  REQUIRED_FIELDS = ['name']
+
+  def __str__(self):
+      return self.email
+
+  def has_perm(self, perm, obj=None):
```

Comment:
Is this used anywhere? Else, we can remove this. And bring it when needed.

Reply:
no that comes in default base user template , I'll remove that.

Reply:
all these methods are required to be kept as it has been written. 
    They are important for providing mandatory functionality to be used with the Admin module.